.github/workflows/create-release.yml

```
@@ -27,10 +27,47 @@ jobs:
           python-version: 3.8
       - name: Change version number
         id: version
+        # See this issue for explanation and testing:
+        # https://github.com/cmu-delphi/delphi-epidata/pull/1473
         run: |
           python -m pip install bump2version
-          echo -n "::set-output name=next_tag::"
-          bump2version --list ${{ github.event.inputs.versionName }} | grep new_version | sed -r s,"^.*=",,
+          allowed_pattern="^(major|minor|patch|[0-9]+\.[0-9]+\.[0-9]+)$"
+
+          if [[ ! ${{ github.event.inputs.versionName }} =~ $allowed_pattern ]]; then
+            echo "\nInvalid version name: ${{ github.event.inputs.versionName }}"
+            exit 1
+          fi
+
+          if [[ ${{ github.event.inputs.versionName }} =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            # use given version number
+            NEXT_TAG="${{ github.event.inputs.versionName }}"
+          else
+            # calculate new version number based on given tag
+            NEXT_TAG=$(bump2version --dry-run --list ${{ github.event.inputs.versionName }} | grep ^new_version | sed -r s,"^.*=",,)
+          fi
```

i dont know why this didnt occur to me when we were looking at #1473, but this is less repetitive and a little tighter:
```suggestion
          if [[ ${{ github.event.inputs.versionName }} =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
            # use given version number
            NEXT_TAG="${{ github.event.inputs.versionName }}"
          elif [[ ${{ github.event.inputs.versionName }} =~ ^(major|minor|patch)$ ]]; then
            # calculate new version number based on given tag
            NEXT_TAG=$(bump2version --dry-run --list ${{ github.event.inputs.versionName }} | grep ^new_version | sed -r s,"^.*=",,)
          else
            echo "\nInvalid version name: ${{ github.event.inputs.versionName }}"
            exit 1
          fi
```